pczt/src/lib.rs

```diff
@@ -1 +1,122 @@
+//!     - Nullifier check (using Jubjub / Pallas arithmetic)
+//!     - KDF plus note decryption (AEAD_CHACHA20_POLY1305)
+//!     - SignatureHash algorithm
+//!     - Signatures (RedJubjub / RedPallas)
+//!     - A source of randomness.
+//! - Combiner (anyone can execute)
+//!   - Combines several PCZTs that represent the same transaction into a single PCZT.
+//! - Spend Finalizer (anyone can execute)
+//!   - Combines partial transparent signatures into `script_sig`s.
+//! - Transaction Extractor (anyone can execute)
+//!   - Creates bindingSig and extracts the final transaction.
 
+use getset::{Getters, MutGetters};
+use serde::{Deserialize, Serialize};
+
+pub mod roles;
+
+mod common;
+pub mod orchard;
+pub mod sapling;
+pub mod transparent;
+
+const MAGIC_BYTES: &[u8] = b"PCZT";
+const PCZT_VERSION_1: u32 = 1;
+
+#[cfg(feature = "zcp-builder")]
+const SAPLING_TX_VERSION: u32 = 4;
+const V5_TX_VERSION: u32 = 5;
+const V5_VERSION_GROUP_ID: u32 = 0x26A7270A;
+
+/// A partially-created Zcash transaction.
+#[derive(Clone, Debug, Serialize, Deserialize, Getters, MutGetters)]
+pub struct Pczt {
+    /// Global fields that are relevant to the transaction as a whole.
+    global: common::Global,
+
+    //
+    // Protocol-specific fields.
+    //
+    // Unlike the `TransactionData` type in `zcash_primitives`, these are not optional.
+    // This is because a PCZT does not always contain a semantically-valid transaction,
+    // and there may be phases where we need to store protocol-specific metadata before
+    // it has been determined whether there are protocol-specific inputs or outputs.
+    //
+    #[getset(get = "pub", get_mut = "pub")]
+    transparent: transparent::Bundle,
+    #[getset(get = "pub", get_mut = "pub")]
+    sapling: sapling::Bundle,
+    #[getset(get = "pub", get_mut = "pub")]
+    orchard: orchard::Bundle,
+}
+
+impl Pczt {
+    /// Parses a PCZT from its encoding.
+    pub fn parse(bytes: &[u8]) -> Result<Self, ParseError> {
+        if bytes.len() < 8 {
+            return Err(ParseError::TooShort);
+        }
+        if &bytes[..4] != MAGIC_BYTES {
+            return Err(ParseError::NotPczt);
+        }
+        let version = u32::from_le_bytes(bytes[4..8].try_into().unwrap());
+        if version != PCZT_VERSION_1 {
+            return Err(ParseError::UnknownVersion(version));
+        }
+
+        // This is a v1 PCZT.
+        postcard::from_bytes(&bytes[8..]).map_err(ParseError::Postcard)
+    }
+
+    /// Serializes this PCZT.
+    pub fn serialize(&self) -> Vec<u8> {
+        let mut bytes = vec![];
+        bytes.extend_from_slice(MAGIC_BYTES);
+        bytes.extend_from_slice(&PCZT_VERSION_1.to_le_bytes());
+        postcard::to_extend(self, bytes).expect("can serialize into memory")
+    }
+}
+
+/// Errors that can occur while parsing a PCZT.
+#[derive(Debug)]
+pub enum ParseError {
+    /// The bytes do not contain a PCZT.
+    NotPczt,
+    /// The PCZT encoding was invalid.
+    Postcard(postcard::Error),
```

Comment:
```suggestion
    Invalid(postcard::Error),
```
The meaning of the error doesn't depend on Postcard, even if the wrapped error type does. A client that only matched on `Invalid(_)`, for example, would not be depending on the encoding being Postcard (or implemented using any particular Postcard library).